api/internal/core/storage/storage.go

```diff
@@ -36,12 +36,13 @@ type WatchResponse struct {
 type Keypair struct {
 	Key   string
 	Value string
+	// If true, this keypair will NOT be processed in store.
+	Skipped bool
```

Comment:
Why we need to skip it in store not storage?

Reply:
@ShiningRush

IMHO, no data processing logic in storage layer would keep it single responsibility, which is only to provide data to store. Store will handle more logic to determine if really going to skip the data during processing. Looking forward to your insights.

Reply:
@ShiningRush  From my point of view, the difference is about the design.

1. Storage provides ALL data to Store for further processing
>  This would lead to skip data in Store, leaving Storage out of data processing
2. Storage provides ONLY VALID data to Store for further processing
> This would guide us to eliminate invalid data before sending to Store, as you mentioned.

Looking forward to your insights.

Reply:
@ShiningRush 

Cause the original skip check is done in Store, I guess your opinion is the design 1. If not, we could also use design 2 instead. It's not a heavy workload. 😄

Reply:
In fact, I prefer option 2, because invalid data should be cleaned from the source. If the skip logic of storage requires store participation, then this part of the content can be provided to the store as options instead of a `Skipped` flag, which is just a notification to the store and does not have much meaning.

BTW: I did not participate in the review of the previous skip logic, so I don't know it before.
Sometimes I’m busy, so I can’t participate in every pr, sorry : (

Reply:
@ShiningRush 
Just reminder, the placeholder could not be cleared, because APISIX needs to simulate a directory like `/apisix/route/`, otherwise there will be problems with `watch`.

Reply:
Got it

Reply:
@ShiningRush

> If the skip logic of storage requires store participation, then this part of the content can be provided to the store as options instead of a `Skipped` flag, which is just a notification to the store and does not have much meaning.

Thanks for your opinion. I would use design 2 to refactor this functionality.

> BTW: I did not participate in the review of the previous skip logic, so I don't know it before.
> Sometimes I’m busy, so I can’t participate in every pr, sorry : (

That's OK. Just try enjoy the open source. 😄